staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -160,7 +202,22 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 	}
 
 	data := originalData
-	if s.yaml {
+	var yamlToJSONError error
+	if s.strict {
+		// In strict mode always pass the data trough the YAMLToJSONStrict converter.
+		// This is done to catch duplicate fields for both JSON and YAML. For JSON data,
+		// the output would equal the input, unless there is a parsing error such as duplicate fields.
+		altered, err := yaml.YAMLToJSONStrict(data)
```

Comment:
would it be cleaner to move this logic down into the strict block below and have it look at `originalData`, and keep the existing gvk extraction path as-is?

Reply:
i think this might be clearer, yes.